swo_aws_extension/flows/fulfillment/base.py

```diff
@@ -1,56 +1,102 @@
+        context (InitialAWSContext): The context of the order.
 
     Returns:
         None
     """
-    logger.info(f'Start processing {order["type"]} order {order["id"]}')
-    context = OrderContext.from_order(order)
+    logger.info(f"Start processing {context.order_type} order {context.order_id}")
     try:
-        if is_purchase_order(order):
-            purchase.run(client, context)
-        elif is_change_order(order):
+        if context.is_purchase_order():
+            purchase_context = PurchaseContext.from_context(context)
+            purchase.run(client, purchase_context)
+        elif context.is_change_order():
             logger.info("Processing change order")
             change_order.run(client, context)
-        elif is_termination_order(order):  # pragma: no branch
-            context = TerminateContext.from_order(order)
+        elif context.is_termination_order():  # pragma: no branch
+            context = TerminateContext.from_context(context)
             terminate.run(client, context)
         else:
-            logger.warning(f"Unsupported order type: {order['type']}")
+            logger.warning(f"Unsupported order type: {context.order_type}")
     except Exception:
         notify_unhandled_exception_in_teams(
             "fulfillment",
-            order["id"],
+            context.order_id,
             strip_trace_id(traceback.format_exc()),
         )
         raise
+
+
+def setup_contexts(mpt_client, orders):
+    """
+    List of contexts from orders
+    Args:
+        mpt_client (MPTClient): MPT client
+        orders (list): List of orders
+
+    Returns: List of contexts
+    """
+    purchase_orders_pls_status_map = {
+        order["id"]: is_partner_led_support_enabled(order)
+        for order in orders
+        if not get_mpa_account_id(order)
+        and get_transfer_type(order)
+        not in [
+            TransferTypesEnum.TRANSFER_WITH_ORGANIZATION,
+            TransferTypesEnum.SPLIT_BILLING,
+        ]
+    }
+    pls_mpa_pool_map = {
+        pls: get_available_mpa_from_pool(pls)
+        for pls in set(purchase_orders_pls_status_map.values())
+    }
+
+    contexts = []
+
+    for order in orders:
+        context = InitialAWSContext(order=order)
+        if context.is_purchase_order():
+            if order["id"] in purchase_orders_pls_status_map:
+                pls_status = purchase_orders_pls_status_map[order["id"]]
+                # Check if there is an available MPA for the order.
+                # If there is, assign it to the context.
+                # Otherwise, leave it as None to be handled on the order processed.
+                if pls_mpa_pool_map[pls_status]:
+                    # Assign first available MPA from the pool
+                    context.airtable_mpa = pls_mpa_pool_map[pls_status].pop(0)
+                    # Add buyer details to the order to obtain the SCU
+                    order["buyer"] = get_buyer(mpt_client, order["buyer"]["id"])
```

Comment:
Not critical, but I would suggest you here to add `select(agreement,agreement.buyer)` to the `get_processing_orders` function